R/LSBuffonsneedlemanipulation.R

```diff
@@ -0,0 +1,166 @@
+    propPost <- dbeta(xValue, a + crosses, b + observations - crosses)
+    propPrior <-dbeta(xValue, a, b)
+    dataProp = data.frame(values = c(xValue, xValue),
+                          density = c(propPost, propPrior),
+                          group = c(rep("Posterior",201), rep("Prior",201))
+    )
+    # axis specification
+    propPlot0 <- ggplot2::ggplot(data = dataProp,  ggplot2::aes(x = values, y = density)) +
+      ggplot2::xlab("Proportion of Crosses") +
+      ggplot2::ylab("Density") #+
+    #ggplot2::coord_cartesian(xlim = c(0, 1), ylim = c(0, 1.6*max(propPost)))
+    
+    # fill in the plot
+    propPlot$plotObject <- jaspGraphs::themeJasp(propPlot0) +
+      ggplot2::geom_line(ggplot2::aes(linetype = group), size = 1) +
+      ggplot2::scale_linetype_manual("", values = c("Posterior" = "solid",
+                                                    "Prior" = "dashed")) 
+    if (legend1){
+      propPlot$plotObject <-  propPlot$plotObject + 
+        ggplot2::theme(legend.position = c(.17, .9))
+    }
+    jaspResults[["propPlot"]] <- propPlot
+  }
+  
+  
+  ## 2. Distribution Plot
+  if (plot3){
+    distPlot <- createJaspPlot(title = paste("Implied Prior and Posterior for", "\u03c0"),  width = 480, height = 320)
+    distPlot$position <- 3
+    distPlot$dependOn(c("k", "n", "a", "b", "length", "CI", "plot3", "CIArrow"))
+    distPlot$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
+    
+    # values
+    x <- seq(2,4,0.01)
+    yPost <- 2 * l / (x^2 * d) * dbeta((2 * l / (x * d)), a + crosses, b + observations - crosses)
+    yPrior <- 2 * l / (x^2 * d) * dbeta((2 * l / (x * d)), a, b)
+    yPi <- seq(0, 1.6*max(yPost), 1.6*max(yPost)/99)
+    
+    data = data.frame(values = c(x, x, rep(pi, 100)),
+                      density = c(yPost, yPrior, yPi),
+                      group = c(rep("Implied Posterior",201), rep("Implied Prior",201), rep("\u03c0", 100))
+    )
+    data$group<-factor(data$group, levels=c("Implied Posterior","Implied Prior","\u03c0"))
+    
+    # axis specification
+    distPlot0 <- ggplot2::ggplot(data = data,  ggplot2::aes(x = values, y = density)) +
+      ggplot2::ggtitle("") + # for , pi
+      ggplot2::xlab("\u03c0") +
+      ggplot2::ylab("Density") +
+      ggplot2::coord_cartesian(xlim = c(2, 4), ylim = c(0, 1.6*max(yPost)))
+    
+    # fill in the plot
+    distPlot$plotObject <- jaspGraphs::themeJasp(distPlot0) +
+      ggplot2::geom_line(ggplot2::aes(color = group, linetype = group), size = 1) +
+      ggplot2::scale_color_manual("", values = c("Implied Posterior" = "black",
+                                                 "Implied Prior" = "black",
+                                                 "\u03c0" = "red")) +
+      ggplot2::scale_linetype_manual("", values = c("Implied Posterior" = "solid",
+                                                    "Implied Prior" = "dashed",
+                                                    "\u03c0" = "solid"))
```

Comment:
when I do this, it crashes
`      #ggplot2::scale_color_manual("", values = c(gettext("Implied Posterior") = "black",
      #                                           gettext("Implied Prior") = "black",
      #                                           gettext("\u03c0") = "red")) +
      #ggplot2::scale_linetype_manual("", values = c(gettext("Implied Posterior") = "solid",
      #                                              gettext("Implied Prior") = "dashed",
      #                                              gettext("\u03c0") = "solid"))`

Reply:
`      #ggplot2::scale_color_manual("", values = c(gettext("Implied Posterior") = "black",
      #                                           gettext("Implied Prior") = "black",
      #                                           gettext("\u03c0") = "red")) +
      #ggplot2::scale_linetype_manual("", values = c(gettext("Implied Posterior") = "solid",
      #                                              gettext("Implied Prior") = "dashed",
      #                                              gettext("\u03c0") = "solid"))`

Reply:
when I do `gettext("Implied Posterior")` it crashes.

Reply:
or do you mean `values = c("Implied Posterior" = gettext("black"),` ?

Reply:
@Kucharssim I don't think that `gettext` is needed here. This code maps characters/ factor levels to colors and line types. As long as the names (`"Implied Posterior"`, etc.) are not translated in the particular `data.frame` it should be fine.

@Jiashun97 if you do need to translate these strings, try something like this:
```r
setNames(c("black", "black", "red"), c(gettext("Implied Posterior"), gettext("Implied Prior"), "\u03c0"))
```
or in 2 lines of code with `values <- c(...); names(values) <- c(gettext(...), ...)`

Reply:
It's needed because the text goes into the legend of the plot, so it should be translated. 

@Jiashun97 either you can do what @vandenman suggested, or rewrite something like this
```r
data = data.frame(values = c(x, x, rep(pi, 100)),
                  density = c(yPost, yPrior, yPi),
                  group = c(rep("posterior",201), rep("prior",201), rep("pi", 100))
)

labels <- c(gettext("Implied Posterior"), gettext("Implied Prior"), "\u03c0")

plot <- ggplot2::ggplot(data = data,  ggplot2::aes(x = values, y = density)) +
  ggplot2::ggtitle("") + # for , pi
  ggplot2::xlab(gettext("\u03c0")) +
  ggplot2::ylab(gettext("Density")) +
  ggplot2::coord_cartesian(xlim = c(2, 4), ylim = c(0, 1.6*max(yPost))) +
  ggplot2::geom_line(ggplot2::aes(color = group, linetype = group), size = 1) +
  ggplot2::scale_color_manual(name = "", 
                              values = c("posterior" = "black",
                                         "prior"     = "black",
                                         "pi"        = "red"), 
                              labels = labels) +
  ggplot2::scale_linetype_manual(name = "", 
                                 values = c("posterior" = "solid",
                                            "prior"     = "dashed",
                                            "pi"        = "solid"),
                                 labels = labels)
plot <- jaspGraphs::themeJasp(plot)
```
that way, the data frame for plotting is using internally untranslated strings for the grouping, but then you use the `labels` argument to replace the strings with the gettexted strings which will be shown then if you enable the legend.

Reply:
I did what @Kucharssim suggested. That would be clearer for me.